Update E-Commerce Store guide

icode247 opened the pull request.

ovaistariq commented on docs/guides/nuxtjs/ecommerce/index.mdx:
```diff
@@ -1,5 +1,5 @@
 ---
-title: "Tutorial: E-Commerce Store"
+title: "Building an E-Commerce Store with Tigris full-text Search"
```
Building an E-Commerce Store with Nuxt and Tigris Full-text Search

icode247 replied:
@ovaistariq fixed

ovaistariq reviewed (commented):
We shouldn't change the image [configure-vercel-project.png](https://github.com/tigrisdata/tigris-docs/pull/293/files#diff-a6bfe8a9403182fb51dc612d3fcf5ea56535ed42b6b9bcf60ae2a47bbd72e6a6). I masked out the secrets in the image.

icode247 commented:
> We shouldn't change the image [configure-vercel-project.png](https://github.com/tigrisdata/tigris-docs/pull/293/files#diff-a6bfe8a9403182fb51dc612d3fcf5ea56535ed42b6b9bcf60ae2a47bbd72e6a6). I masked out the secrets in the image.

I don't think this image was used anyw

> We shouldn't change the image [configure-vercel-project.png](https://github.com/tigrisdata/tigris-docs/pull/293/files#diff-a6bfe8a9403182fb51dc612d3fcf5ea56535ed42b6b9bcf60ae2a47bbd72e6a6). I masked out the secrets in the image.

Those images were not used anywhere in this guide.

icode247 commented:
> We shouldn't change the image [configure-vercel-project.png](https://github.com/tigrisdata/tigris-docs/pull/293/files#diff-a6bfe8a9403182fb51dc612d3fcf5ea56535ed42b6b9bcf60ae2a47bbd72e6a6). I masked out the secrets in the image.

Those images were not used anywhere in this guide.

ovaistariq commented:
> > We shouldn't change the image [configure-vercel-project.png](https://github.com/tigrisdata/tigris-docs/pull/293/files#diff-a6bfe8a9403182fb51dc612d3fcf5ea56535ed42b6b9bcf60ae2a47bbd72e6a6). I masked out the secrets in the image.
> 
> Those images were not used anywhere in this guide.

It looks like in your diff you are adding the image.

leggetter commented:
Is this duplicate content of https://www.tigrisdata.com/blog/ecommerce-store-with-tigris-search/ ?

Looks like this is also causing us confusion. See https://github.com/tigrisdata/tigris-docs/pull/293#issuecomment-1409412228

So, we'll be getting penalized for duplicate content unless we set a canonical link between the original and the duplicate.

IMO we should delete one of the pieces of content. Or, change one considerably. Based on our priorities, I'd vote for deleting one.

icode247 commented:
> Is this duplicate content of https://www.tigrisdata.com/blog/ecommerce-store-with-tigris-search/ ?
> 
> Looks like this is also causing us confusion. See [#293 (comment)](https://github.com/tigrisdata/tigris-docs/pull/293#issuecomment-1409412228)
> 
> So, we'll be getting penalized for duplicate content unless we set a canonical link between the original and the duplicate.
> 
> IMO we should delete one of the pieces of content. Or, change one considerably. Based on our priorities, I'd vote for deleting one.

I noticed a huge similarity in both contents. The main author replicated the guide on the blog. I support we remove one too.

ovaistariq commented:
> > Is this duplicate content of https://www.tigrisdata.com/blog/ecommerce-store-with-tigris-search/ ?
> > Looks like this is also causing us confusion. See [#293 (comment)](https://github.com/tigrisdata/tigris-docs/pull/293#issuecomment-1409412228)
> > So, we'll be getting penalized for duplicate content unless we set a canonical link between the original and the duplicate.
> > IMO we should delete one of the pieces of content. Or, change one considerably. Based on our priorities, I'd vote for deleting one.
> 
> I noticed a huge similarity in both contents. The main author replicated the guide on the blog. I support we remove one too.

How about removing it from the blog and then adding a redirect from blog to the docs?

leggetter commented:
There are some code differences and the blog version doesn't callout pre-loading data - this is instead handled in the setup script. But I don't see the setup script run anywhere. Whatever version we choose we need to run through the steps to ensure that the app works.

The biggest difference in the content I can see is that the blog version sets up the project in the Console and the docs version uses the CLI.

I have reviewed the blog version and updated it to emphasize the tech used (Tigris Search, Netlify Functions, and Nuxt) so I'm more comfortable with the content in the blog version. Though, I haven't run through the tutorial steps to ensure the code shown works as described.

I've created a fresh PR that updates the docs version to be the contents of the blog version with an additional callout for  the CLI project creation approach:
https://github.com/tigrisdata/tigris-docs/pull/294

I've created a PR for the blog to update the blog point to provide some overview information and point to the guide in the docs https://github.com/tigrisdata/blog/pull/67 Though, we can setup a redirect and remove this post if preferred.

icode247 commented:
> There are some code differences and the blog version doesn't callout pre-loading data - this is instead handled in the setup script. But I don't see the setup script run anywhere. Whatever version we choose we need to run through the steps to ensure that the app works.
> 
> The biggest difference in the content I can see is that the blog version sets up the project in the Console and the docs version uses the CLI.
> 
> I have reviewed the blog version and updated it to emphasize the tech used (Tigris Search, Netlify Functions, and Nuxt) so I'm more comfortable with the content in the blog version. Though, I haven't run through the tutorial steps to ensure the code shown works as described.
> 
> I've created a fresh PR that updates the docs version to be the contents of the blog version with an additional callout for the CLI project creation approach: #294
> 
> I've created a PR for the blog to update the blog point to provide some overview information and point to the guide in the docs [tigrisdata/blog#67](https://github.com/tigrisdata/blog/pull/67) Though, we can setup a redirect and remove this post if preferred.

The blog version looks great now. But there few things I need to point out on PR (294)(https://github.com/tigrisdata/tigris-docs/pull/294) 
- The command to clone the project, clones the older version of Tigris code (I updated it in[ PR](Update E-Commerce Store guide #293) ).
- The code work, I have tested it and followed the rest of the tutorial and confirmed it works.
- I tried the local CLI project creation flow works but does not establish a connection with the Tigris service running locally on my computer.

leggetter commented:
> The command to clone the project, clones the older version of Tigris code (I updated it in[ PR](Update E-Commerce Store guide https://github.com/tigrisdata/tigris-docs/pull/293) ).

Can we copy the updated code over?

> I tried the local CLI project creation flow works but does not establish a connection with the Tigris service running locally on my computer.

Is there a problem here https://github.com/tigrisdata/tigris-docs/pull/294/files#r1091861769 ?

What problem do you see? Can you share the output?

icode247 commented:
> > The command to clone the project, clones the older version of Tigris code (I updated it in[ PR](Update E-Commerce Store guide #293) ).
> 
> Can we copy the updated code over?
> 
> > I tried the local CLI project creation flow works but does not establish a connection with the Tigris service running locally on my computer.
> 
> Is there a problem here https://github.com/tigrisdata/tigris-docs/pull/294/files#r1091861769 ?
> 
> What problem do you see? Can you share the output?

I have tested the commands, it works as expected. But as I pointed out the need to change the clone command to:
```
git clone https://github.com/tigrisdata/tigris-netlify-ecommerce
```
To get the lasted version of Tigris code.
Then change the command `tigris create application ecommerce_search "search tutorial"` to `tigris create project ecommerce_search "search tutorial"` that all.

leggetter commented:
> To get the lasted version of Tigris code.

But I saw a few differences in the code between the blog and the guide. Should those changes also be moved over?

icode247 commented:
> > To get the lasted version of Tigris code.
> 
> But I saw a few differences in the code between the blog and the guide. Should those changes also be moved over?

All changes made to the code were to update them to the lasted Tigris code syntax. Yes, we should move them over.

ovaistariq commented:
I would prefer we remove the CLI step of project creation and do it the same way we do in the blog through the web console.

ovaistariq reviewed (commented):
Let's make sure your branch is up to date with `main` and the change to file `static/img/screenshots/configure-vercel-project.png` in this diff needs to be removed.

leggetter commented:
> I would prefer we remove the CLI step of project creation and do it the same way we do in the blog through the web console.

The CLI usage is hidden in an expandable section. However, I'd actually prefer and recommend the CLI first approach.

A piece of feedback I've not yet converted into a ticket is that we should provide a clearer CLI driven DX/journey. The vast majority of tools and dev workflows, especially in a JS world, are from the CLI so we should provide two journeys with great DX. The act of getting the many Tigris project settings/config into an app alone is good reason for a CLI.

I'll write up some suggestions on the CLI shortly.

Is there a reason we should hide the CLI?

leggetter commented:
We should close this PR and be working from the branch created for https://github.com/tigrisdata/tigris-docs/pull/293 which contains the ported blog content.

ovaistariq commented:
> > I would prefer we remove the CLI step of project creation and do it the same way we do in the blog through the web console.
> 
> The CLI usage is hidden in an expandable section. However, I'd actually prefer and recommend the CLI first approach.
> 
> A piece of feedback I've not yet converted into a ticket is that we should provide a clearer CLI driven DX/journey. The vast majority of tools and dev workflows, especially in a JS world, are from the CLI so we should provide two journeys with great DX. The act of getting the many Tigris project settings/config into an app alone is good reason for a CLI.
> 
> I'll write up some suggestions on the CLI shortly.
> 
> Is there a reason we should hide the CLI?

Here is my reasoning:
- Most of our users are new, which will come through the flow: website/docs/blog > signup > web console > create a project. We can better track the journey this way rather than going through the CLI, and the user would already be used to creating the project on the web and running the `create-tigris-app` command (which is a CLI). Adding tracking to CLIs is frowned upon, while tracking on the web is something people are used to.
- The tigris CLI is going to go through major structural changes to the command layout. I am also looking at other changes to the CLI to make it lean.

leggetter commented:
@ovaistariq 

> Most of our users are new, which will come through the flow: website/docs/blog > signup > web console > create a project. We can better track the journey this way rather than going through the CLI, and the user would already be used to creating the project on the web and running the create-tigris-app command (which is a CLI). Adding tracking to CLIs is frowned upon, while tracking on the web is something people are used to.

Obviously a fine line. And our values/principles/behaviours will guide us in our decision.

My opinion is that signup should be via the web interface and not driven by the CLI. The fallout benefit from that is that you can see the developer journey prior to signup. However, once signed up we should have two journeys:

1. A UI-driven journey
2. A CLI-driven journey

The problem with the `tigris-create-app` approach is it's too restrictive. I need a CLI so I can integrate Tigris into my preferred developer workflow. I have a way I create apps and I just want Tigris to do the Tigris things such as create a project and manage my Tigris account. I also want to use the CLI in CI/CD workflows.

> The tigris CLI is going to go through major structural changes to the command layout. I am also looking at other changes to the CLI to make it lean.

I don't think this in on our [roadmap](https://www.notion.so/tigrisdata/Tigris-Platform-Roadmap-e685db595b9e4771aeb950496e764a6b) (internal link), but it should be.
